src/index.ts

```diff
@@ -176,16 +205,33 @@ export async function gru<MasterArgs = undefined>(
             // TODO: remove excess workers
         }
     }
+    function resizeDedicatedWorkers() {
+        if (!options.dedicatedWorkers) {
+            return
+        }
 
-    function startWorker() {
+        Object.keys(options.dedicatedWorkers).map((workerName) => {
+            if (workerMap[workerName]) {
+                logger.debug({ workerName }, `dedicated worker already exists`)
+                return
+            }
+            startWorker(workerName)
+        })
+    }
+
+    function startWorker(workerName?: string) {
```

Comment:
When would you want to omit `workerName`?

Reply:
the generic workers dont have a name